x-pack/plugins/ml/server/routes/data_frame_analytics.ts

```diff
@@ -285,12 +324,77 @@ export function dataFrameAnalyticsRoutes({ router, mlLicense }: RouteInitializat
+        let destinationIndex: string | undefined;
+        const analyticsJobDeleted: DeleteDataFrameAnalyticsWithIndexStatus = { success: false };
+        const targetIndexDeleted: DeleteDataFrameAnalyticsWithIndexStatus = { success: false };
+        const targetIndexPatternDeleted: DeleteDataFrameAnalyticsWithIndexStatus = {
+          success: false,
+        };
+
+        // Check if analyticsId is valid and get destination index
+        if (deleteTargetIndex || deleteIndexPattern) {
+          try {
+            const dfa = await context.ml!.mlClient.callAsCurrentUser('ml.getDataFrameAnalytics', {
+              analyticsId,
+            });
+            if (Array.isArray(dfa.data_frame_analytics) && dfa.data_frame_analytics.length > 0) {
+              destinationIndex = dfa.data_frame_analytics[0].dest.index;
+            }
+          } catch (e) {
+            return response.customError(wrapError(e));
           }
-        );
+
+          // If user checks box to delete the destinationIndex associated with the job
+          if (destinationIndex && deleteTargetIndex) {
+            // Verify if user has privilege to delete the destination index
+            const userCanDeleteDestIndex = await userCanDeleteIndex(context, destinationIndex);
+            // If user does have privilege to delete the index, then delete the index
+            if (userCanDeleteDestIndex) {
+              try {
+                await context.ml!.mlClient.callAsCurrentUser('indices.delete', {
+                  index: destinationIndex,
+                });
+                targetIndexDeleted.success = true;
+              } catch (deleteIndexError) {
+                targetIndexDeleted.error = wrapError(deleteIndexError);
+              }
+            } else {
+              return response.forbidden();
+            }
+          }
+
+          // Delete the index pattern if there's an index pattern that matches the name of dest index
+          if (destinationIndex && deleteIndexPattern) {
+            try {
+              const indexPatternId = await getIndexPatternId(context, destinationIndex);
+              if (indexPatternId) {
+                await deleteIndexPatternById(context, indexPatternId);
+              }
+              targetIndexPatternDeleted.success = true;
+            } catch (deleteIndexPatternError) {
+              targetIndexPatternDeleted.error = wrapError(deleteIndexPatternError);
+            }
+          }
+        }
+        // Grab the target index from the data frame analytics job id
+        // Delete the data frame analytics
+        let results;
+
+        try {
+          results = await context.ml!.mlClient.callAsCurrentUser('ml.deleteDataFrameAnalytics', {
+            analyticsId,
+          });
+          analyticsJobDeleted.success = true;
+        } catch (deleteDFAError) {
+          analyticsJobDeleted.error = wrapError(deleteDFAError);
+        }
+        results = {
+          ...results,
```

Comment:
the contents of `results` does not need to be returned now as `analyticsJobDeleted` has been added

Reply:
Removed in https://github.com/elastic/kibana/pull/66934/commits/5bd97e9c466ad1a88d32da4f432583d5ddc3aed0